pkg/reconciler/pipelinerun/affinity_assistant_test.go

```diff
@@ -514,3 +612,39 @@ spec:
 		})
 	}
 }
+
+func seedTestData(d test.Data, resource, verb string) (context.Context, Reconciler, func()) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+
+	c := Reconciler{
+		KubeClientSet: fakek8s.NewSimpleClientset(),
+		Images:        pipeline.Images{},
+	}
+	for _, s := range d.StatefulSets {
+		c.KubeClientSet.AppsV1().StatefulSets(s.Namespace).Create(ctx, s, metav1.CreateOptions{})
+	}
+	for _, n := range d.Nodes {
+		c.KubeClientSet.CoreV1().Nodes().Create(ctx, n, metav1.CreateOptions{})
+	}
+	for _, p := range d.Pods {
+		c.KubeClientSet.CoreV1().Pods(p.Namespace).Create(ctx, p, metav1.CreateOptions{})
+	}
+	var reaction testing2.ReactionFunc
```

Comment:
The way we do this in existing pipelines tests is just to return the clientset, and then the test case can add a fake reactor if needed. I think using the same pattern here would probably be easier to understand.